cupyx/scipy/sparse/data.py

```diff
@@ -73,6 +73,35 @@ def count_nonzero(self):
         """
         return cupy.count_nonzero(self.data)
 
+    def mean(self, axis=None, dtype=None, out=None):
+        """Compute the arithmetic mean along the specified axis.
+
+        Args:
+            axis (int or ``None``): Axis along which the sum is computed.
+                If it is ``None``, it computes the average of all the elements.
+                Select from ``{None, 0, 1, -2, -1}``.
+
+        Returns:
+            cupy.ndarray: Summed array.
+
+        .. seealso::
+           :meth:`scipy.sparse.spmatrix.mean`
+
+        """
+        validateaxis(axis)
+        nRow, nCol = self.shape
+        data = self.data.copy()
+        n = 1
+
+        if not axis:
+            n = nRow * nCol
+        if axis == 0 or axis == -2:
+            n = nRow
+        else:
+            n = nCol
+
+        return self._with_data(data / n).sum(axis, dtype, out).flatten()
```

Comment:
```suggestion
        return self._with_data(data / n).sum(axis, dtype, out)
```

- `flatten` changes `ndim` if `axis=None`.
- It seems `flatten` is verbose otherwise.

Reply:
The current mean for ndarray returns a row for any axis, however if I don't add flatten, this function will return a column if the axis is 1.